# Symmetry
Il s'agit dans ce document d'étudier l'effet de 3 éléments de symétrie différents pour 3 atomes du cristal.

In [75]:
from pymatgen import Structure
from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
from pymatgen.analysis.bond_valence import BVAnalyzer
from pymatgen.core.composition import Composition
from pymatgen.ext.matproj import MPRester
from jupyter_jsmol.pymatgen import quick_view
from pymatgen.core.operations import SymmOp

In [76]:
with MPRester("SmhzvzkiqMXecCh0o4") as m:

    # Structure de mon cristal, GaSb
    struct = m.get_structure_by_material_id("mp-1018059")

In [77]:
view = quick_view(struct, conventional = True)

In [78]:
display(view)

JsmolView(layout=Layout(align_self='stretch', height='400px'))

Le groupe d'espace de ce cristal est $P6_{3}mc$, grâce aux deux premiers éléments ($P6$), nous pouvons déduire le système primitif hexagonal du crital. Les différentes types de symétrie qui composent ce groupe d'espace sont :

* Rotation hélicoïdale $6_3$
* Plan miroir m
* Plan de glissement c

# Atome n°2 : réflexion
Symétrie 10 : $(x, y, z) \rightarrow (-y, -x, z)$

In [85]:
#obtention des coordonnées de l'atome numéro 2
print("Coordonnées atome 2 :" + str((BVAnalyzer().get_oxi_state_decorated_structure(struct)[3].a,BVAnalyzer().get_oxi_state_decorated_structure(struct)[3].b,BVAnalyzer().get_oxi_state_decorated_structure(struct)[3].c)))
#matrice de l'opération de symétrie correspondante
print("Matrice de l'opération de symétrie : \n" + str(SymmOp.from_xyz_string("-y,-x,z")))

Coordonnées atome 2 :(0.333333, 0.666667, 0.374773)
Matrice de l'opération de symétrie : 
Rot:
[[ 0. -1.  0.]
 [-1.  0.  0.]
 [ 0.  0.  1.]]
tau
[0. 0. 0.]


In [86]:
view = quick_view(struct, conventional = True)
display(view)
view.script('draw symop 10 {atomno=2}')

JsmolView(layout=Layout(align_self='stretch', height='400px'))

Cette symétrie est définie par un plan miroir. Il est représenté par un plan d'après quoi l'atome va effectuer une symétrie orthogonale. Ce plan (-210) est un plan miroir noté m et définit une opération de réflexion, nous le retrouvons bien dans la description du groupe spatial.

Si nous partons des coordonnées générales de l'atome ; (x, y, z), l'application de la symétrie orthogonale va mener à l'atome en (-y, -x, z).

Si l'on prend l'atome 2, qui est un atome de Sb et dont les coordonnées sont $(0.333333, 0.666667, 0.374773)$, l'application de la symétrie nous mène à la position $(-0.666667, -0.333333, 0.374773)$, en revenant dans la maille conventionnelle, on obtient $(0.333333, 0.666667, 0.374773)$, soit la propre image de l'atome 2.

Cet opérateur de réflexion est définit par la matrice suivante:

$$\hat{\sigma} = \begin{pmatrix} 0 & -1 & 0 \\ -1 & 0 & 0 \\ 0 & 0 & 1 \end{pmatrix}$$

# Atome n°3 : rotation hélicoïdale
Symétrie 4 : $(x, y, z) \rightarrow (-x, -y, z+1/2)$

In [89]:
#obtention des coordonnées de l'atome numéro 3 
print("Coordonnées atome 3 :" + str((BVAnalyzer().get_oxi_state_decorated_structure(struct)[0].a,BVAnalyzer().get_oxi_state_decorated_structure(struct)[0].b,BVAnalyzer().get_oxi_state_decorated_structure(struct)[0].c)))
#matrice de l'opération de symétrie correspondante
print("Matrice de l'opération de symétrie : \n" + str(SymmOp.from_xyz_string("-x,-y,z+1/2")))

Coordonnées atome 3 :(0.666667, 0.333333, 0.500127)
Matrice de l'opération de symétrie : 
Rot:
[[-1.  0.  0.]
 [ 0. -1.  0.]
 [ 0.  0.  1.]]
tau
[0.  0.  0.5]


In [82]:
view = quick_view(struct, conventional = True)
display(view)
view.script('draw symop 4 {atomno=3}')

JsmolView(layout=Layout(align_self='stretch', height='400px'))

Cette symétrie est représentée par un élément de symétrie jouant deux rôles. D'une part, on observe un axe de rotation autour duquel gravite une flèche courbe en rouge, ceci nous indique premièrement la présence d'une rotation de 180°. D'autre part, cet axe rotation joue également le rôle de vecteur de translation, sa direction est la même que celle du vecteur de base $\bf{c}$ et sa norme vaut la moitié de c. Cet opération de symétrie est appelée rotation hélicoïdale $2_1$ et est tout à fait compatible avec la rotation hélicoïdale $6_3$ donnée dans le groupe spatial, en effet, $2_1$ correspond à $6_3$ appliquée 3 fois. 

Si nous partons des coordonnées générales de l'atome ; (x, y, z), dans un premier temps, l'application de la rotation de 180° autour de l'axe va mener l'atome en (-x, -y, z) et dans un deuxième temps, la translation va le conduire en (-x, -y, z+1/2). 

Si l'on part de l'atome 3, qui est un atome de Ga et dont les coordonnées sont $(0.666667, 0.333333, 0.500127)$, l'application de la symétrie nous mène à la position $(-0.666667, -0.333333, 1.000127)$, en revenant dans la maille conventionnelle, on obtient $(0.333333, 0.666667 , 0.000127)$, soit les coordonnées de l'atome 1.

Cet opérateur est définit par la matrice de rotation et par le vecteur de translation suivants:

$$\hat{C}_{2} = \begin{pmatrix} -1 & 0 & 0 \\ 0 & -1 & 0 \\ 0 & 0 & 1 \end{pmatrix} \;\;\;\;\;\;\; \tau = \begin{pmatrix} 0 \\ 0 \\ 1/2 \end{pmatrix}$$



# Atome n°4 : réflexion avec glissement
Symétrie 7 : (x, y, z) -> (y, x, z+1/2)

In [90]:
#obtention des coordonnées de l'atome numéro 4 
print("Coordonnées atome 4 :" + str((BVAnalyzer().get_oxi_state_decorated_structure(struct)[2].a,BVAnalyzer().get_oxi_state_decorated_structure(struct)[2].b,BVAnalyzer().get_oxi_state_decorated_structure(struct)[2].c)))
#matrice de l'opération de symétrie correspondante
print("Matrice de l'opération de symétrie : \n" + str(SymmOp.from_xyz_string("y,x,z+1/2")))

Coordonnées atome 4 :(0.666667, 0.333333, 0.874773)
Matrice de l'opération de symétrie : 
Rot:
[[0. 1. 0.]
 [1. 0. 0.]
 [0. 0. 1.]]
tau
[0.  0.  0.5]


In [84]:
view = quick_view(struct, conventional = True)
display(view)
view.script('draw symop 7 {atomno=4}')

JsmolView(layout=Layout(align_self='stretch', height='400px'))

Cette symétrie est représentée par un plan de glissement, d'une part, on observe un plan bleu correspondant à ce plan de glissement comprenant les vecteurs $\bf{b}$ et $\bf{c}$, soit le plan (100), l'atome subit ensuite une translation selon la moitié du vecteur $\bf{c}$, parallèlement au plan de glissement. Cet élément de symétrie nous donne une opération de réflexion avec glissement. Comme le glissement se fait selon la moitié du vecteur de base c, il est dit axial. Cette symétrie est notée $c$, nous la retrouvons en dernière lettre dans la description du groupe spatial.

Si nous partons des coordonnées générale de l'atome ; (x, y, z), dans un premier temps, l'application de la réflexion via le plan de glissement va mener l'atome en (y, x, z) et dans un deuxième temps, le glissement va le conduire en (y, x, z+1/2). 

Si l'on part de l'atome 4, qui est un atome de Sb et dont les coordonnées sont $(0.666667, 0.333333, 0.874773)$, l'application de la symétrie nous mène à la position $(0.333333, 0.666667 , 1.374773)$, en revenant dans la maille conventionnelle, on obtient $(0.333333, 0.666667 , 0.374773)$, soit les coordonnées de l'atome 2.


Cet opérateur est définit par la matrice de rotation et par le vecteur de translation suivants:

$$\hat{\sigma} = \begin{pmatrix} 0 & 1 & 0 \\ 1 & 0 & 0 \\ 0 & 0 & 1 \end{pmatrix} \;\;\;\;\;\;\; \tau = \begin{pmatrix} 0 \\ 0 \\ 1/2 \end{pmatrix}$$